cli/pcluster/config/config_patch.py

```diff
@@ -225,17 +225,12 @@ def check(self):
         patch_allowed = True
 
         for change in self.changes:
-            if change.update_policy != UpdatePolicy.IGNORED:
+            check_result, reason, action_needed, print_change = change.update_policy.check(change, self)
+            if print_change:
```

Comment:
Should we put this change right before `rows.append(`? As it is I think this would imply if we do not print change the `patch_allow` is always True?

Reply:
Yes, if no changes are found to be displayed the patch should be allowed. In that case we'd have only ALLOWED, IGNORED and/or READ_ONLY not blocking changes. Do you think I am missing anything?

Reply:
Ok, so I think this logic makes sense with what we are implementing today, but it is a bit weird that we would couple `print_change` with `patch_allowed`, when I think `patch_allowed ` should only be associated with `check_result`. It's a minor thing, but may be worth doing just to avoid confusions in the future